content/blog/2024/grid-masonry-nuance.md

```diff
@@ -0,0 +1,480 @@
+duplicating the track definitions:
+
+```css
+/* grid fallback… duplicate properties */
+.grid-fallback {
+  display: grid;
+  grid-template-columns: repeat(auto-fit, minmax(200px, 1fr));
+
+  display: masonry;
+  masonry-template-tracks: repeat(auto-fit, minmax(200px, 1fr));
+}
+```
+
+We could put the track definition inside a variable if we want,
+but it still requires explicit duplication.
+Because we can rely on various properties working
+in only one layout mode,
+and we get a simple mode toggle,
+it's fairly simple to set up any fallback we want.
+But the trade off is that we don't have
+the the most obvious fallback _built in_ from the start.
+
+While I was experimenting with different fallback paths,
+I was trying to maintain the spirit of masonry in some way --
+tightly packed columns --
+with as little effort as possible.
+My favorite solution was the default fallback
+provided by grid-integrated masonry,
+with the addition of rough aspect-ratio attributes when possible --
+e.g. `is-short` and `is-tall`.
+All items span multiple rows,
+but short items span fewer rows,
+and tall items span more rows.
+
+<figure>
+  {{ embed.codepen(
+    id='wvVKvrN',
+    title='Rough Puff Masonry',
+    user='miriamsuzanne'
+  ) }}
+  <figcaption>
+    I borrowed the markup and images
+    from Jen Simmons,
+    and I'm pretty happy with the result.
+  </figcaption>
+</figure>
+
+It's not a perfect solution,
+but I was surprised how well it works.
+This is a damn good fallback story in my mind.
+Of course, _effort_ here is hard to measure.
+If we know the size of each item,
+and have access to the markup,
+then this is pretty straight-forward.
+But those are big _ifs_.
+
+On my personal site,
+image galleries fit this description perfectly.
+Maybe I'll start using it.
+
+It's also fair to say that fallbacks
+become less essential over time.
+Looking further down the road,
+we might also want to consider…
+
+## What corners are we painting ourselves into?
+
+Frankly, it's hard to know.
+But this is one of the questions
+that the Chrome team keeps bringing up.
+If we attach masonry to grid now,
+they can never diverge in the future.
+Every future addition to grid
+will need to take masonry into account.
+
+While I can see features
+that might build on a masonry precedent,
+I don't actually have a clear picture of what
+might _conflict_ with masonry-in-grid down the road.
+I don't see obvious issues
+that are likely to come up --
+but Chrome is right that it could
+theoretically become an issue.
+
+How much do we worry about that now?
+I don't know.
+
+## What's next?
+
+Join the conversation!
+There are several threads
+where developers can leave feedback…
+
+- on the [original proposal](https://github.com/w3c/csswg-drafts/issues/10233)
+- or the [alternate syntax](https://github.com/w3c/csswg-drafts/issues/9041)
+
+It's most helpful to talk about your own use-cases,
+and how you expect to use this feature
+(if you do).
+I avoid arguments about…
```

Comment:
I like this gentle reminder of what is less helpful if people want to share their opinions. Helps to keep things focused.